### ISSUE TITLE
amp-story: Status messages are not explicitly announced by assistive technologies

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#32517]
# Status messages are not explicitly announced by assistive technologies

## WCAG Level
Level AA

## Priority
Medium

## Pages/screens/components affected
* [AMP Story Player - AMP Story 360](https://www.gstatic.com/amphtml/stamp/qa/amp-story-360.html)

## Description
When status messages/notifications appear on a page dynamically, without automatically receiving keyboard focus, it is important that these messages also be communicated correctly to assistive technologies.

![The instruction overlay for the 360 AMP Story](https://user-images.githubusercontent.com/54933393/107253825-a0939b00-6a2e-11eb-9954-e510773cc3a7.png)

In [AMP Story Player - AMP Story 360](https://www.gstatic.com/amphtml/stamp/qa/amp-story-360.html) an instruction overlay is shown on the second page, which disappears after a few seconds. This overlay is not announced by assistive technologies.

## User impact
People using assistive technologies such as screen readers and screen magnification tools may not be aware when status messages are injected into the page outside of their current area of focus. For example, if a confirmation message only appears visually (without being explicitly announced) as a "toast" style notification, or even as a permanently visible new notification block in a specific part of the screen that is not currently in focus, screen reader users may not even be aware that a confirmation message appeared at all.

## Required solution
Whenever status messages or notifications are injected in the page, and these status messages do not receive focus (which is usually not appropriate, since this moves the user's context and interrupts their current activity), make sure that the messages are coded in such a way that they are announced by assistive technologies. For web-based content, this typically means identifying the status message as an [ARIA live region](https://www.w3.org/TR/wai-aria-1.1/#attrs_liveregions) - by either using a [live region `role`](https://www.w3.org/TR/wai-aria-1.1/#live_region_roles), or adding an [`aria-live` attribute](https://www.w3.org/TR/wai-aria-1.1/#aria-live).

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Implementation guidance

The current overlay is already set up as an ARIA live region, using `aria-live="polite"`. However, the actual text content "Move device to explore" is hidden using `aria-hidden="true"`.

```
<div class="i-amphtml-story-360-discovery" aria-live="polite">
  <div class="i-amphtml-story-360-discovery-animation"></div>
  <span class="i-amphtml-story-360-discovery-text" aria-hidden="true">Move device to explore</span>
</div>
```

Remove the `aria-hidden="true"` from the `<span>`, which should ensure that the overlay is correctly announced by assistive technologies when it appears.

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue:
1. Turn on your screen reader, and open a page that includes a status message.
2. Activate the component that triggers the status message.
3. Make sure that screen readers announce appropriate feedback when the status message visibly appears on the page.

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website has been tested for the same issue.
* All issues identified throughout the website have been resolved or filed as new issues.

## Related standards
[WCAG 2.1 Success Criterion 4.1.3 Status Messages (Level A)](https://www.w3.org/TR/WCAG21/#status-messages)

## More information
* [W3C - Understanding Success Criterion 4.1.3: Status Messages](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html)
* [Knowbility - Exploring WCAG 2.1: 4.1.3 Status Messages](https://knowbility.org/blog/2018/WCAG21-413StatusMessages/)
* [MDN web docs - ARIA live regions](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions)

## Test data
Test date: January 2021
